### PR TITLE
fix avatar

### PR DESCRIPTION
**Pull request by codengoo**

### CODE REVIEW

File: apps/mobile/src/app/screens/home/homedrawer/components/AvatarMessage/index.tsx [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-	textAvatar: string;
+	id: string;
+	username: string;
 	isShow: boolean;
 }
-export const AvatarMessage = React.memo(({ isShow, onPress, avatar, textAvatar }: IProps) => {
+export const AvatarMessage = React.memo(({ isShow, onPress, id, username }: IProps) => {
 	const { themeValue } = useTheme();
 	const styles = style(themeValue);
+	const userById = useSelector(selectMemberByUserId(id || ''));
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/components/AvatarMessage/index.tsx]: get avatar from message

File: apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx [hunk 2]
```diff
@@ -283,8 +283,9 @@ const MessageItem = React.memo((props: MessageItemProps) => {
 				<View style={[styles.wrapperMessageBox, !isCombine && styles.wrapperMessageBoxCombine]}>
 					<AvatarMessage
 						onPress={onPressAvatar}
-						avatar={message?.isMe ? userProfile?.user?.avatar_url : message?.user?.avatarSm}
-						textAvatar={isDM ? message?.display_name || message?.user?.username : message?.user?.username}
+						id={message?.user?.id}
+						avatar={message?.avatar}
+						username={isDM ? message?.display_name || message?.user?.username : message?.user?.username}
```
Comment (sontung-nccvn) [inline apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx]: refactor

File: apps/mobile/src/app/screens/home/homedrawer/components/UserProfile/index.tsx [hunk 3]
```diff
@@ -83,14 +83,13 @@ const UserProfile = React.memo(({ userId, user, onClose, checkAnonymous, message
 		<View style={[styles.wrapper]}>
 			<View style={[
 				styles.backdrop,
-				{ backgroundColor: user?.avatar_url || user?.avatarSm ? color : Colors.titleReset }
+				{ backgroundColor: user?.avatar_url || user?.avatarSm || userById?.user?.avatar_url ? color : Colors.titleReset }
```
Comment (sontung-nccvn) [inline apps/mobile/src/app/screens/home/homedrawer/components/UserProfile/index.tsx]: duplicate

File: apps/mobile/src/app/temp-ui/MezonAvatar/index.tsx [hunk 4]
```diff
@@ -49,18 +49,17 @@ const MezonAvatar = React.memo((props: IMezonAvatarProps) => {
...
+			<View style={[styles.boxImage, { height, width }, isBorderBoxImage && styles.borderBoxImage]}>
+				<MezonClanAvatar
+					alt={username}
+					image={avatarUrl}
+					lightMode
+				/>
 			</View>
-		</TouchableWithoutFeedback>
+
+			{userStatus && <View style={[styles.statusCircle, userStatus ? styles.online : styles.offline]} />}
```
Comment (sontung-nccvn) [inline apps/mobile/src/app/temp-ui/MezonAvatar/index.tsx]: !!userStatus

File: apps/mobile/src/app/screens/home/homedrawer/components/UserProfile/index.tsx [hunk 5]
```diff
@@ -83,14 +83,13 @@ const UserProfile = React.memo(({ userId, user, onClose, checkAnonymous, message
...
 			]}>
 				<View style={[styles.userAvatar]}>
 					<MezonAvatar
 						width={80}
 						height={80}
-						// avatarUrl={userById?.user?.avatar_url || user?.avatar_url || user?.avatarSm}
-						avatarUrl={user?.avatar_url || user?.avatarSm}
+						avatarUrl={userById?.user?.avatar_url || user?.avatar_url || user?.avatarSm}
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/components/UserProfile/index.tsx]: not use userById

**Review by phuocnguyenncc — APPROVED** [review]

**Review by phuocnguyenncc — CHANGES_REQUESTED** [review]
revert keep using userById in UserProfile

File: apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx [hunk 6]
```diff
@@ -211,6 +211,10 @@ const MessageItem = React.memo((props: MessageItemProps) => {
 		return message?.clan_nick || message?.user?.username || (checkAnonymous ? 'Anonymous' : message?.username);
 	}, [checkAnonymous, message?.clan_nick, message?.user?.username, message?.username, message?.display_name, isDM]);
 
+	const usernameMessage = useMemo(() => {
+		return isDM ? message?.display_name || message?.user?.username : message?.user?.username
+	}, [isDM, message]);
```
Comment (phuocnguyenncc) [inline apps/mobile/src/app/screens/home/homedrawer/MessageItem.tsx]: dependency should [isDM, message?.display_name, message?.user?.username]);

**Review by phuocnguyenncc — APPROVED** [review]

**Pull request merged**